File: BackofficeSpringboot/src/main/java/itu/mbds/tpt/entity/Post.java
package itu.mbds.tpt.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "post")
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private Utilisateur utilisateur;

    @Column(name = "created_at", nullable = false, updatable = false)
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime createdAt;

    @Column(name = "titre", nullable = false)
    private String titre;

    @Column(name = "longitude")
    private double longitude;

    @Column(name = "latitude")
    private double latitude;

    @Column(name = "description", nullable = false)
    private String description;

    @Column(name = "status", nullable = false)
    private boolean status;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PostDetail> postDetails;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }


}

File: ClientWinForm/Models/Post.cs
﻿using System;
using System.Collections.Generic;

namespace ClientWinForm.Models
{
    public class Post
    {
        public int poste_id { get; set; }
        public int user_id { get; set; }
        public DateTime CreatedAt { get; set; }
        public string Titre { get; set; }
        public string Longitude { get; set; }
        public string Latitude { get; set; }
        public string Description { get; set; }
        public bool Status { get; set; }
        public Utilisateur Utilisateur { get; set; }
        public List<PostDetail> Postedetails { get; set; }
        public List<Echange> Echanges { get; set; } // Nouvelle propriété pour les échanges
    }
}
